pandas/_libs/join.pyx

```diff
@@ -478,7 +494,7 @@ def inner_join_indexer(ndarray[numeric_object_t] left, ndarray[numeric_object_t]
             if lval == rval:
                 lindexer[count] = i
                 rindexer[count] = j
-                result[count] = rval
+                result[count] = lval
```

Comment:
why this change, this didn't break anything?

Reply:
lval == rval here, this just makes this block match whats done elsewhere